ddocs/medic-client/views/total_clinics_by_facility/map.js

```diff
@@ -1,33 +1,20 @@
 function(doc) {
   if (doc.type === 'clinic') {
-    var dh = doc.parent && doc.parent.parent,
-        dh_id = dh && dh._id,
-        dh_name = dh && dh.name,
-        hc_id = doc.parent && doc.parent._id,
-        hc_name = doc.parent && doc.parent.name,
-        cl_name = doc.name,
-        phone = doc.contact && doc.contact.phone,
-        placeId = doc.place_id || (doc.contact && doc.contact.rc_code);
 
-    if (!cl_name && doc.contact) {
-      if (doc.contact.name) {
-        cl_name = doc.contact.name;
-        if (placeId) {
-          cl_name += ' ' + placeId;
-        }
-      } else {
-        cl_name = placeId;
+    var districtId = doc.parent && doc.parent.parent && doc.parent.parent._id,
+        healthCenterId = doc.parent && doc.parent._id;
+
+    emit([ districtId, healthCenterId, doc._id, 0 ]);
+    if (doc.contact && doc.contact._id) {
+      emit([ districtId, healthCenterId, doc._id, 1 ], { _id: doc.contact._id });
+    }
+    var index = 2;
+    var parent = doc.parent;
+    while(parent) {
+      if (parent._id) {
+        emit([ districtId, healthCenterId, doc._id, index++ ], { _id: parent._id });
       }
+      parent = parent.parent;
```

Comment:
It saves us from having `null` elements in the lineage. I don't think we really support parents with no ID this is mostly being defensive because you really don't want to have an exception in a view map function...